File: adapters/csharp/Urirun.cs
// urirun — C# SDK for building urirun.bindings.v2 documents (no dependencies).
using System.Collections.Generic;

namespace Urirun
{
    public sealed class Connector
    {
        public const string BindingsVersion = "urirun.bindings.v2";
        private readonly string _id;
        private readonly string _scheme;
        private readonly string _target;
        private readonly List<string> _members = new List<string>();

        public Connector(string id, string scheme, string target = "host")
        {
            _id = id;
            _scheme = scheme;
            _target = target;
        }

        // schemaJson and argvJson must be valid JSON fragments.
        public Connector Command(string route, string schemaJson, string argvJson)
        {
            var uri = $"{_scheme}://{_target}/{route}";
            _members.Add($"\"{uri}\":{{\"uri\":\"{uri}\",\"kind\":\"command\",\"adapter\":\"argv-template\",\"inputSchema\":{schemaJson},\"argv\":{argvJson},\"meta\":{{\"connector\":\"{_id}\"}},\"policy\":{{\"allowExecute\":true}}}}");
            return this;
        }

        public string BindingsJson() =>
            $"{{\"version\":\"{BindingsVersion}\",\"bindings\":{{{string.Join(",", _members)}}}}}";
    }
}

File: adapters/java/Urirun.java
// urirun — Java SDK for building urirun.bindings.v2 documents (no dependencies).
// The caller passes valid JSON fragments for the schema and argv; the SDK
// assembles the standardized document.
import java.util.ArrayList;
import java.util.List;

public final class Urirun {
    public static final String BINDINGS_VERSION = "urirun.bindings.v2";

    public static final class Connector {
        private final String id;
        private final String scheme;
        private final String target;
        private final List<String> members = new ArrayList<>();

        public Connector(String id, String scheme) { this(id, scheme, "host"); }

        public Connector(String id, String scheme, String target) {
            this.id = id;
            this.scheme = scheme;
            this.target = target;
        }

        public Connector command(String route, String schemaJson, String argvJson) {
            String uri = scheme + "://" + target + "/" + route;
            members.add("\"" + uri + "\":{\"uri\":\"" + uri
                + "\",\"kind\":\"command\",\"adapter\":\"argv-template\",\"inputSchema\":" + schemaJson
                + ",\"argv\":" + argvJson + ",\"meta\":{\"connector\":\"" + id
                + "\"},\"policy\":{\"allowExecute\":true}}");
            return this;
        }

        public String bindingsJson() {
            return "{\"version\":\"" + BINDINGS_VERSION + "\",\"bindings\":{"
                + String.join(",", members) + "}}";
        }
    }
}
